Orhu/Summer2023Project
# implemented enemy stat singleton

Enemies now grab some stats from the enemy stat singleton:
- Movement speed
- Health
- Damage (on touch multiplier disabled but can be easily enabled if we want it)

Additionally, I went ahead and added the EventSystem under the HUD prefab for organization.

To test:
Mess with values in enemy stat manager. Although no ranged attacks are on enemies yet, you can modify the player's damage multiplier in PlayerController.cs manually to see that the projectile damage multiplier behaves correctly from the player perspective (and enemies use the same behavior). Health and move speed both should be pretty easy to test as-is. If you feel like it, you can also uncomment the line in BaseStateMachine to enable on touch damage scaling and see that the multiplier works for on touch damage.
